src/schema-status-bar-item.ts

```diff
@@ -0,0 +1,165 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Red Hat, Inc. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import {
+  ExtensionContext,
+  window,
+  commands,
+  StatusBarAlignment,
+  TextEditor,
+  StatusBarItem,
+  QuickPickItem,
+  ThemeColor,
+  workspace,
+} from 'vscode';
+import { CommonLanguageClient, RequestType } from 'vscode-languageclient/node';
+
+type FileUri = string;
+interface JSONSchema {
+  name?: string;
+  description?: string;
+  uri: string;
+}
+
+interface MatchingJSONSchema extends JSONSchema {
+  usedForCurrentFile: boolean;
+  fromStore: boolean;
+}
+
+interface SchemaItem extends QuickPickItem {
+  schema?: MatchingJSONSchema;
+}
+
+// eslint-disable-next-line @typescript-eslint/ban-types
+const getJSONSchemas: RequestType<FileUri, MatchingJSONSchema[], {}> = new RequestType('yaml/get/all/jsonSchemas');
+
+// eslint-disable-next-line @typescript-eslint/ban-types
+const getSchema: RequestType<FileUri, JSONSchema[], {}> = new RequestType('yaml/get/jsonSchema');
+
+export let statusBarItem: StatusBarItem;
+
+let client: CommonLanguageClient;
+export function createJSONSchemaStatusBarItem(context: ExtensionContext, languageclient: CommonLanguageClient): void {
+  if (statusBarItem) {
+    updateStatusBar(window.activeTextEditor);
+    return;
+  }
+  const commandId = 'yaml.select.json.schema';
+  client = languageclient;
+  commands.registerCommand(commandId, () => {
+    return showSchemaSelection();
+  });
+  statusBarItem = window.createStatusBarItem(StatusBarAlignment.Right);
+  statusBarItem.command = commandId;
+  context.subscriptions.push(statusBarItem);
+
+  context.subscriptions.push(window.onDidChangeActiveTextEditor(updateStatusBar));
+  setTimeout(() => updateStatusBar(window.activeTextEditor), 5000);
+}
+
+async function updateStatusBar(editor: TextEditor): Promise<void> {
+  if (editor && editor.document.languageId === 'yaml') {
+    // get schema info there
+    const schema = await client.sendRequest(getSchema, editor.document.uri.toString());
+    if (schema.length === 0) {
+      statusBarItem.text = 'No JSON Schema';
+      statusBarItem.tooltip = 'Select JSON Schema';
+      statusBarItem.backgroundColor = undefined;
+    } else if (schema.length === 1) {
+      statusBarItem.text = schema[0].name ?? schema[0].uri;
+      statusBarItem.tooltip = 'Select JSON Schema';
+      statusBarItem.backgroundColor = undefined;
+    } else {
+      statusBarItem.text = 'Multiple JSON Schemas...'; //schema.map((val) => val.name ?? val.uri).join(' | ');
+      statusBarItem.tooltip = 'Multiple JSON Schema used to validate this file, click to select one';
+      statusBarItem.backgroundColor = new ThemeColor('statusBarItem.errorBackground');
```

Comment:
Is this an error situation or a warning one? Should we use `warningBackground` instead?